editor/gui/editor_quick_open_dialog.h

```diff
@@ -0,0 +1,226 @@
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef EDITOR_QUICK_OPEN_DIALOG_H
+#define EDITOR_QUICK_OPEN_DIALOG_H
+
+#include "core/templates/oa_hash_map.h"
+#include "scene/gui/dialogs.h"
+
+class Button;
+class CenterContainer;
+class CheckButton;
+class EditorFileSystemDirectory;
+class LineEdit;
+class HFlowContainer;
+class MarginContainer;
+class PanelContainer;
+class ScrollContainer;
+class StringName;
+class Texture2D;
+class TextureRect;
+class VBoxContainer;
+
+class QuickOpenResultItem;
+
+enum class QuickOpenDisplayMode {
+	GRID,
+	LIST,
+};
+
+class QuickOpenResultContainer : public VBoxContainer {
+	GDCLASS(QuickOpenResultContainer, VBoxContainer)
+
+public:
+	void init(const Vector<StringName> &p_base_types);
+	void handle_search_box_input(const Ref<InputEvent> &p_ie);
+	void update_results(const String &p_query);
+
+	bool has_nothing_selected() const;
+	String get_selected() const;
+
+	void cleanup();
+
+	QuickOpenResultContainer();
+	~QuickOpenResultContainer();
+
+protected:
+	void _notification(int p_what);
+
+private:
+	static const int TOTAL_ALLOCATED_RESULT_ITEMS = 100;
+	static const int SHOW_ALL_FILES_THRESHOLD = 30;
+
+	struct Candidate {
+		String file_name;
+		String file_directory;
+
+		Ref<Texture2D> thumbnail;
+		float score = 0;
+	};
+
+	Vector<StringName> base_types;
+	Vector<Candidate> candidates;
+
+	String query;
+	int selection_index = -1;
+	int num_visible_results = 0;
+	int max_total_results = 0;
+
+	QuickOpenDisplayMode content_display_mode = QuickOpenDisplayMode::LIST;
+	Vector<QuickOpenResultItem *> result_items;
+
+	ScrollContainer *scroll_container = nullptr;
+	VBoxContainer *list = nullptr;
+	HFlowContainer *grid = nullptr;
+
+	PanelContainer *panel_container = nullptr;
+	CenterContainer *no_results_container = nullptr;
+	Label *no_results_label = nullptr;
+
+	Label *file_details_path = nullptr;
+	Button *display_mode_toggle = nullptr;
+	CheckButton *include_addons_toggle = nullptr;
+
+	OAHashMap<StringName, Ref<Texture2D>> file_type_icons;
+
+	static QuickOpenDisplayMode get_adaptive_display_mode(const Vector<StringName> &p_base_types);
+
+	void _create_initial_results(bool p_include_addons);
+	void _find_candidates_in_folder(EditorFileSystemDirectory *p_directory, bool p_include_addons);
+
+	int _sort_candidates(const String &p_query);
+	void _update_result_items(const int p_new_visible_results_count, int p_new_selection_index);
+
+	void _move_selection_index(const Key p_key);
```

Comment:
```suggestion
	void _move_selection_index(Key p_key);
```
No `const` for simple types